code/modules/food_and_drinks/food/snacks_other.dm

```diff
@@ -18,46 +19,55 @@
 	filling_color = "#FFD700"
 	list_reagents = list("nutriment" = 3, "vitamin" = 1)
 	tastes = list("cheese" = 1)
+	foodtype = DAIRY
 
 /obj/item/weapon/reagent_containers/food/snacks/watermelonslice
 	name = "watermelon slice"
 	desc = "A slice of watery goodness."
 	icon_state = "watermelonslice"
 	filling_color = "#FF1493"
 	tastes = list("watermelon" = 1)
+	foodtype = FRUIT
 
 /obj/item/weapon/reagent_containers/food/snacks/candy_corn
 	name = "candy corn"
 	desc = "It's a handful of candy corn. Can be stored in a detective's hat."
+	junkiness = 20
 	icon_state = "candy_corn"
 	list_reagents = list("nutriment" = 4, "sugar" = 2)
 	filling_color = "#FF8C00"
 	tastes = list("candy corn" = 1)
+	foodtype = JUNKFOOD
 
 /obj/item/weapon/reagent_containers/food/snacks/chocolatebar
 	name = "chocolate bar"
 	desc = "Such, sweet, fattening food."
+	junkiness = 20
 	icon_state = "chocolatebar"
 	list_reagents = list("nutriment" = 2, "sugar" = 2, "cocoa" = 2)
 	filling_color = "#A0522D"
 	tastes = list("chocolate" = 1)
+	foodtype = JUNKFOOD
 
 /obj/item/weapon/reagent_containers/food/snacks/hugemushroomslice
 	name = "huge mushroom slice"
 	desc = "A slice from a huge mushroom."
 	icon_state = "hugemushroomslice"
 	list_reagents = list("nutriment" = 3, "vitamin" = 1)
 	tastes = list("mushroom" = 1)
+	foodtype = VEGETABLES
 
 /obj/item/weapon/reagent_containers/food/snacks/popcorn
 	name = "popcorn"
 	desc = "Now let's find some cinema."
+	junkiness = 4
```

Comment:
Don't assign junkiness to items the chef makes.

The whole point of the junkiness var is to make food that is in vendors really crappy and non-satisfying and to encourage them to go to the chef. The chef's food should never be worse than what's in the vendors.